kafka-connect-gcp-storage/src/it/scala/io/lenses/streamreactor/connect/gcp/storage/utils/GCPProxyContainerTest.scala

```diff
@@ -0,0 +1,90 @@
+package io.lenses.streamreactor.connect.gcp.storage.utils
+
+import cats.implicits.toBifunctorOps
+import com.google.cloud.storage.BucketInfo
+import com.google.cloud.storage.Storage
+import com.typesafe.scalalogging.LazyLogging
+import io.lenses.streamreactor.connect.cloud.common.config.ConnectorTaskId
+import io.lenses.streamreactor.connect.cloud.common.config.TaskIndexKey
+import io.lenses.streamreactor.connect.cloud.common.sink.CloudPlatformEmulatorSuite
+import io.lenses.streamreactor.connect.gcp.storage.auth.GCPStorageClientCreator
+import io.lenses.streamreactor.connect.gcp.storage.config.GCPConfigSettings._
+import io.lenses.streamreactor.connect.gcp.storage.config.AuthMode
+import io.lenses.streamreactor.connect.gcp.storage.config.AuthModeSettingsConfigKeys
+import io.lenses.streamreactor.connect.gcp.storage.config.GCPConfig
+import io.lenses.streamreactor.connect.gcp.storage.config.UploadConfigKeys
+import io.lenses.streamreactor.connect.gcp.storage.sink.GCPStorageSinkTask
+import io.lenses.streamreactor.connect.gcp.storage.storage.GCPStorageFileMetadata
+import io.lenses.streamreactor.connect.gcp.storage.storage.GCPStorageStorageInterface
+import io.lenses.streamreactor.connect.testcontainers.GCPStorageContainer
+import io.lenses.streamreactor.connect.testcontainers.PausableContainer
+import io.lenses.streamreactor.connect.testcontainers.TestContainersPausableContainer
+
+import java.io.File
+import java.nio.file.Files
+import scala.util.Try
+
+trait GCPProxyContainerTest
+    extends CloudPlatformEmulatorSuite[GCPStorageFileMetadata, GCPStorageStorageInterface, GCPStorageSinkTask, Storage]
+    with TaskIndexKey
+    with AuthModeSettingsConfigKeys
+    with UploadConfigKeys
+    with LazyLogging {
+
+  implicit val connectorTaskId: ConnectorTaskId = ConnectorTaskId("unit-tests", 1, 1)
+
+  override val container: PausableContainer = new TestContainersPausableContainer(GCPStorageContainer())
+
+  override val prefix: String = "connect.gcpstorage"
+
+  override def createStorageInterface(client: Storage): GCPStorageStorageInterface =
```

Comment:
Is there an advantae to capture the exception go to an either and  fail if it's a Left? Since we don't handle the error let it bubble. ideally to avoid the Java approach we can return an Either or even a Try and let the caller handle it

Reply:
No advantage.  I like the second approach you suggest.  The code as it is is ultimately a mixture of styles so not easy to get consistency without a lot of further work.  I've made some tweaks to improve the situation a bit.